src/server/dfly_main.cc

```diff
@@ -439,6 +440,93 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
+bool InsideContainer() {
```

Comment:
I think it would be nice to respect cgroups limit in general, not only container-wide limits.

We can query the process' cgroup by parsing `/proc/self/cgroup` and then we can check `/sys/fs/cgroup/{process_cgroup_hierarchy}/memory.max` to get a more granular limit if it was configured.

Reply:
How does it work for a process inside a docker container? what do you mean by `{process_cgroup_hierarchy}` ?

Reply:
Roy, great idea!

Running inside a Docker container, the cgroup of a process (unless specified otherwise) is the system itself. In other words, this is the 'root cgroup' -- which has path `/`.

In cgroup v2, printing the file `/proc/self/cgroup` results in an output of:

```shell
0::<path>
```
and as such, the directory `/sys/fs/cgroup/<path>/` is the process' cgroup parameters.

Reply:
I recommend also taking a look at the generic userspace library 'libcgroup' that can retrieve cgroup parameters instead of parsing text files.

Reply:
Thanks. I looked at libcgroup, and it does seem to do what we need, however it also seems to do much more (e.g., create/modify cgroups) and overall it seems an overkill for this task.